packages/wonder-blocks-birthday-picker/src/components/birthday-picker.js

```diff
@@ -252,63 +267,99 @@ export default class BirthdayPicker extends React.Component<Props, State> {
+                    <OptionItem
+                        key={String(CUR_YEAR - yearOffset)}
+                        label={String(CUR_YEAR - yearOffset)}
+                        value={String(CUR_YEAR - yearOffset)}
+                    />
+                ))}
+            </SingleSelect>
+        );
+    }
 
+    render(): React.Element<any> {
         return (
             <>
                 <View testId="birthday-picker" style={{flexDirection: "row"}}>
-                    <SingleSelect
-                        disabled={disabled}
-                        placeholder={this.labels.month}
-                        onChange={this.handleMonthChange}
-                        selectedValue={month}
-                        style={{minWidth: 110}}
-                        testId="birthday-picker-month"
-                    >
-                        {moment.monthsShort().map((month, i) => (
-                            <OptionItem
-                                key={month}
-                                label={month}
-                                value={String(i)}
-                            />
-                        ))}
-                    </SingleSelect>
-                    <Strut size={Spacing.xSmall_8} />
-                    <SingleSelect
-                        disabled={disabled}
-                        placeholder={this.labels.day}
-                        onChange={this.handleDayChange}
-                        selectedValue={day}
-                        style={{minWidth: 100}}
-                        testId="birthday-picker-day"
-                    >
-                        {Array.from(Array(31)).map((_, day) => (
-                            <OptionItem
-                                key={String(day + 1)}
-                                label={String(day + 1)}
-                                value={String(day + 1)}
-                            />
-                        ))}
-                    </SingleSelect>
+                    {this.renderMonth()}
+
+                    {this.maybeRenderDay()}
+
                     <Strut size={Spacing.xSmall_8} />
-                    <SingleSelect
-                        disabled={disabled}
-                        placeholder={this.labels.year}
-                        onChange={this.handleYearChange}
-                        selectedValue={year}
-                        style={{minWidth: 110}}
-                        testId="birthday-picker-year"
-                    >
-                        {Array.from(Array(120)).map((_, yearOffset) => (
-                            <OptionItem
-                                key={String(CUR_YEAR - yearOffset)}
-                                label={String(CUR_YEAR - yearOffset)}
-                                value={String(CUR_YEAR - yearOffset)}
-                            />
-                        ))}
-                    </SingleSelect>
+
+                    {this.renderYear()}
                 </View>
```

Comment:
This is not i18n friendly. I think long term we need to support date formats. In most countries outside the US, the day comes before the month.